libs/designsystem/shared/floating/src/floating.directive.spec.ts

```diff
@@ -0,0 +1,463 @@
+  });
+
+  describe('input', () => {
+    describe('reference', () => {
+      it('should set', () => {
+        directive.reference = component.floatingElementRef;
+        expect(directive.reference).toEqual(component.floatingElementRef);
+      });
+
+      it('should not add event listeners when only reference is set without triggers', () => {
+        directive.triggers = null;
+        directive.reference = component.floatingElementRef;
+        expect(directive['eventListeners']).toHaveLength(0);
+      });
+    });
+
+    describe('placement', () => {
+      it('should set', () => {
+        const placement: Placement = 'top-start';
+        directive.placement = placement;
+        expect(directive['_placement']).toEqual(placement);
+      });
+    });
+
+    describe('strategy', () => {
+      it('should set', () => {
+        const strategy: Strategy = 'absolute';
+        directive.strategy = strategy;
+        expect(directive['_strategy']).toEqual(strategy);
+      });
+    });
+
+    describe('triggers', () => {
+      it('should set', () => {
+        const eventTriggers: Array<TriggerEvent> = ['hover'];
+        directive.triggers = eventTriggers;
+        expect(directive['_triggers']).toEqual(eventTriggers);
+      });
+
+      it('should not add event listeners when only triggers is set without reference', () => {
+        directive.reference = null;
+        directive.triggers = ['hover'];
+        expect(directive['eventListeners']).toHaveLength(0);
+      });
+
+      it('should add event listeners for click event when reference and triggers is set', () => {
+        directive.triggers = ['click'];
+        directive.reference = component.floatingElementRef;
+        expect(directive['eventListeners']).toHaveLength(1);
+      });
+
+      it('should add event listeners for hover event when reference and triggers is set', () => {
+        directive.triggers = ['hover'];
+        directive.reference = component.floatingElementRef;
+        expect(directive['eventListeners']).toHaveLength(2);
+      });
+
+      it('should add event listeners for click event when reference and triggers is set', () => {
+        directive.triggers = ['focus'];
+        directive.reference = component.floatingElementRef;
+        expect(directive['eventListeners']).toHaveLength(2);
+      });
+    });
+  });
+
+  describe('methods', () => {
+    describe('ngOnInit', () => {
+      let autoUpdateFuncSpy;
+      let computePositionFuncSpy;
+
+      beforeEach(() => {
+        autoUpdateFuncSpy = jasmine.createSpy('autoUpdate').and.returnValue(null);
+        spyOnProperty(floatingUi, 'autoUpdate', 'get').and.returnValue(autoUpdateFuncSpy);
+
+        computePositionFuncSpy = jasmine
+          .createSpy('computePosition')
+          .and.returnValue(new Promise(null));
+        spyOnProperty(floatingUi, 'computePosition', 'get').and.returnValue(computePositionFuncSpy);
+      });
+
+      describe('add styling', () => {
```

Comment:
I believe this describe is redundant, and could be removed for readablility so it reads:
`FloatingDirective method ngOnInit should add style XYZ` instead of `FloatingDirective method ngOnInit add style should add style XYZ`